app/assets/javascripts/controllers/dialog_user/dialog_user_controller.js

```diff
@@ -1,30 +1,54 @@
-ManageIQ.angular.app.controller('dialogUserController', ['API', 'dialogFieldRefreshService', 'miqService', 'dialogUserSubmitErrorHandlerService', 'dialogId', 'apiSubmitEndpoint', 'apiAction', 'finishSubmitEndpoint', 'cancelEndpoint', 'resourceActionId', 'targetId', 'targetType', 'realTargetType', 'openUrl', '$http', '$window', 'dialogReplaceData', 'DialogData', function(API, dialogFieldRefreshService, miqService, dialogUserSubmitErrorHandlerService, dialogId, apiSubmitEndpoint, apiAction, finishSubmitEndpoint, cancelEndpoint, resourceActionId, targetId, targetType, realTargetType, openUrl, $http, $window, dialogReplaceData, DialogData) {
+ManageIQ.angular.app.controller('dialogUserController', ['API', 'dialogFieldRefreshService', 'miqService', 'dialogUserSubmitErrorHandlerService', 'dialogId', 'apiSubmitEndpoint', 'apiAction', 'finishSubmitEndpoint', 'cancelEndpoint', 'resourceActionId', 'targetId', 'targetType', 'realTargetType', 'openUrl', '$http', '$window', 'dialogReplaceRequestId', 'DialogData', function(API, dialogFieldRefreshService, miqService, dialogUserSubmitErrorHandlerService, dialogId, apiSubmitEndpoint, apiAction, finishSubmitEndpoint, cancelEndpoint, resourceActionId, targetId, targetType, realTargetType, openUrl, $http, $window, dialogReplaceRequestId, DialogData) {
   var vm = this;
 
   vm.$onInit = function() {
-    var apiCall = new Promise(function(resolve) {
-      var url = '/api/service_dialogs/' + dialogId +
-        '?resource_action_id=' + resourceActionId +
-        '&target_id=' + targetId +
-        '&target_type=' + targetType;
-
-      resolve(API.get(url, {expand: 'resources', attributes: 'content'}).then(init));
-    });
-
-    Promise.resolve(apiCall).then(miqService.refreshSelectpicker);
+    var url = '/api/service_dialogs/' + dialogId +
+      '?resource_action_id=' + resourceActionId +
+      '&target_id=' + targetId +
+      '&target_type=' + targetType;
+
+    API.get(url, {expand: 'resources', attributes: 'content'})
+      .then(function (data) { vm._dialog = data.content[0] })
+      .then(loadServiceRequest)
+      .then(postprocess)
+      .then(function () {
+        vm.dialog = vm._dialog;
+        delete vm._dialog;
+        vm.dialogLoaded = true;
+      })
+      .then(miqService.refreshSelectpicker);
```

Comment:
I dug a little deeper in the codebase and basically this refresh is called too soon, would be okay to use a setTimeout here? I know it's a hackety hack :confused: @himdel

Reply:
Nah, it's causing more issues than it fixes :disappointed:

Reply:
What else is it supposed to wait for?

Reply:
But this should not be necessary anymore, if the component is using `miq-select`, refresh should happen automatically...

Reply:
Somehow the tagControl is behaving weird, even if I comment out the rewrite of the tags after the API calls, the selects for tags are rendered like this:
![Screenshot from 2019-11-12 15-25-28](https://user-images.githubusercontent.com/649130/68679715-b337cf80-0560-11ea-9e18-edac92ccb362.png)

i.e. if you set the data with a delay, it doesn't get rendered properly with selectpicker